### PR TITLE
Add pre-commit hook for updating of license headers

### PR DESCRIPTION
**Pull request by DomFijan**
## Description
To simplify the updates of license headers we can use the pre-commit hook.
 
## Motivation and Context
Ease maintenance burden, keeps license up to date.

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/coxeter/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/coxeter/blob/master/ContributorAgreement.md).
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/coxeter/blob/master/ChangeLog.txt) and the [credits](https://github.com/glotzerlab/coxeter/blob/master/doc/source/credits.rst).

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 BSD 3-Clause License for coxeter
 
-Copyright (c) 2015-2021 The Regents of the University of Michigan All rights reserved.
+Copyright (c) 2015-2024 The Regents of the University of Michigan All rights reserved.
```
Comment (joaander) [inline LICENSE:3]:
```suggestion
Copyright (c) 2015-2024 The Regents of the University of Michigan. All rights reserved.
```

**Review by joaander — APPROVED** [review]
Thanks! This looks good after a small change.

**Comment by joaander** [comment]
pre-commit.ci autofix

File: LICENSE [hunk 2]
```diff
@@ -1,6 +1,6 @@
 BSD 3-Clause License for coxeter
 
-Copyright (c) 2015-2021 The Regents of the University of Michigan All rights reserved.
```
[the remark this answers is not in the document]
Comment (janbridley) [reply LICENSE:3]: Just curious as to how the start dates are defined!

[the remark this answers is not in the document]
Comment (joaander) [reply LICENSE:3]: Good point. I incorrectly assumed that 2015 was good.

The first date in the range should match the first public release. The last date should match the most recent change. The first commit on coxeter was made in 2015. The first PyPI release was in 2020. I don't remember if coxeter code was made public before then, so 2015 is a fine conservative estimate.

[the remark this answers is not in the document]
Comment (janbridley) [reply LICENSE:3]: Thank you!

**Pull request merged**